### ISSUE TITLE
Jooby Apps composition

### ISSUE THREAD

**Issue by walkman90** [jooby-project/jooby#1835]
Hi, Edgar

I have a question regarding JoobyApp composition in jooby2,  as it behaves slightly differently now.
Let's say we have main app

```java
public class App extends Jooby {
  {
    install(new OpenAPIModule());
    mvc(new Controller());

    use("api/v1/pets", new PetsApp());
  }
}
```

and main app is using `PetsApp`

```java
public class PetsApp extends Jooby {
    {
        onStarting(() -> {
           // won't be executed
        });

        onStarted(() -> {
            // won't be executed
        });

        install(new MyExtension()); // skipped if lateinit == true

        mvc(PetsController.class);
    }
}
```

as marked in a snippet in `PetsApp`: 
- startup callbacks won't be executed
- Extensions with late initialization property `on` won't be initialized

is it expected behavior? is it by design in Jooby2 ?

**Comment by jknack** [comment]
App composition only takes routes from existing app (javadoc mention this).

Why?

1. we should be able to run PetsApp as standalone app (so here your startup callbacks/modules work)
2. if you ship this same PetsApp within a parent/root app, the entire setup is moved to root/parent app

So the goal is more or less be able to run composable app (petsapp) as standalone or include their routes as part of main app (main app must provide all the dependencies for child app)

**Comment by jknack** [comment]
we probably should add more "import options"..., what do you think?

**Comment by walkman90** [comment]
Yeah, it would be nice to have an option similar to Jooby 1, where we were able to import entire app saving all startup procedures and not only routes. It will really help to smoothly migrate from Jooby 1 to Jooby 2

**Comment by apaulau** [comment]
Hi @jknack,

I think I'm facing a similar issue: I'm using Guice, so when I require some service in a child module, the registry can't provide it (Jooby throws RegistryException).

Simplified examples (kotlin):
```kotlin
class App : Kooby({
    install(GuiceModule())
    use(ChildApp())
})

class ChildApp : Kooby({
    get("/child") { require(Service::class).foo() } // throws RegistryException
})

class Service {
    fun foo() {}
}
````

While it works just fine when everything is in one module.
```kotlin
class App : Kooby({
    install(GuiceModule())
    get("/child") { require(Service::class).foo() }
})

class Service {
    fun foo() {}
}
````

I guess #1905 will solve it, right? Am I missing some other solutions?

Thanks.

**Comment by jknack** [comment]
Right, #1905 will be the solution.

**Comment by jknack** [comment]
Released in 2.9.4 as #1905
